conda_libmamba_solver/solver.py

```diff
@@ -451,23 +458,28 @@ def _specs_to_tasks_add(self, in_state: SolverInputState, out_state: SolverOutpu
                         break
 
             # ## Low-prio task ###
-            if name in out_state.conflicts and name not in protected:
+            if name in out_state.conflicts and name not in (*protected, *in_state.requested):
                 tasks[("DISFAVOR", api.SOLVER_DISFAVOR)].append(spec_str)
                 tasks[("ALLOWUNINSTALL", api.SOLVER_ALLOWUNINSTALL)].append(spec_str)
 
             if installed is not None:
-                # ## Regular task ###
-                key = "UPDATE", api.SOLVER_UPDATE
+                installed_spec_str = self._spec_to_str(installed.to_match_spec())
+                if spec_str == installed_spec_str and name not in out_state.conflicts:
+                    # if we just want to force things to STAY as they are, better leave them alone
+                    if name in protected:
+                        key = "LOCK", api.SOLVER_LOCK
+                    else:
+                        key = "ADD_PIN", api.SOLVER_NOOP
+                    spec_str = spec_str.split("::")[1]  # remove channel, otherwise lock/pin fails
 
                 # ## Protect if installed AND history
                 if name in protected:
-                    installed_spec = self._spec_to_str(installed.to_match_spec())
-                    tasks[("USERINSTALLED", api.SOLVER_USERINSTALLED)].append(installed_spec)
+                    tasks[("USERINSTALLED", api.SOLVER_USERINSTALLED)].append(installed_spec_str)
                     # This is "just" an essential job, so it gets higher priority in the solver
                     # conflict resolution. We do this because these are "protected" packages
                     # (history, aggressive updates) that we should try not messing with if
                     # conflicts appear
-                    key = ("UPDATE | ESSENTIAL", api.SOLVER_UPDATE | api.SOLVER_ESSENTIAL)
+                    key = f"{key[0]} | ESSENTIAL", key[1] | api.SOLVER_ESSENTIAL
```

Comment:
I think this line changes

```
('UPDATE | ESSENTIAL', ...) ['ca-certificates', 'certifi', 'openssl'] 
```

to

```
('INSTALL | ESSENTIAL', 131328) ['ca-certificates', 'certifi', 'openssl'] 
```

mamba does

```
('UPDATE ', ...) ['ca-certificates', 'certifi', 'openssl'] 
```

It seems like either way we get the same result.